# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1231209234 18000
#      Mon Jan 05 21:33:54 2009 -0500
# Node ID dd52e541418b6ad9d788c0710bdf12727f06f1ab
# Parent  c3ac9f2772cd3aa2b6248fcf79b5e1520e337a4c
scripts/mkdoc: set defaults for FIND and PERL; check usage

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-01-05  John W. Eaton  <jwe@octave.org>
+
+	* mkdoc: Set defaults for FIND and PERL.  Check usage.
+
 2009-01-01  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* miscellaneous/__xzip__.m: Fix error messages, add tests.
 	
 2008-12-26  Thorsten Meyer  <thorsten.meyier@gmx.de>
 
 	* general/int2str.m, general/num2str.m, strings/base2dec.m,
 	strings/blanks.m, strings/cstrcat.m, strings/findstr.m,
diff --git a/scripts/mkdoc b/scripts/mkdoc
--- a/scripts/mkdoc
+++ b/scripts/mkdoc
@@ -15,20 +15,27 @@
 # for more details.
 # 
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <http://www.gnu.org/licenses/>.
 
 set -e
 
-if test $# != 1; then
+FIND=${FIND:-'find'}
+
+PERL=${PERL:-'perl'}
+
+if test $# -eq 1; then
+  d=$1
+elif test $# -eq 0; then
   d=.
 else
-  d=$1
+  echo "usage: mkdoc [directory]" 1>&2
+  exit 1
 fi
 
 if test -f gethelp; then
   cat << EOF
 ### DO NOT EDIT!
 ###
 ### This file is generated automatically from the Octave sources.
 ### Edit those files instead and run make to update this file.
